client/payment-details/payment-method/index.js

```diff
@@ -11,14 +11,21 @@ import { Card, CardBody, CardHeader } from '@wordpress/components';
  */
 import Loadable from 'components/loadable';
 import CardDetails from './card';
+import CardPresentDetails from './card-present';
 import GiropayDetails from './giropay';
 import SepaDetails from './sepa';
 import SofortDetails from './sofort';
 
 const PaymentDetailsPaymentMethod = ( { charge = {}, isLoading } ) => {
-	let PaymentMethodDetails = CardDetails;
+	let PaymentMethodDetails = <></>;
 	if ( charge.payment_method_details && charge.payment_method_details.type ) {
 		switch ( charge.payment_method_details.type ) {
+			case 'card':
+				PaymentMethodDetails = CardDetails;
+				break;
+			case 'card_present':
+				PaymentMethodDetails = CardPresentDetails;
+				break;
```

Comment:
How about replacing whole switch/case with lookup object of format `{ 'type': Component }` this object is static and can be defined at module level.

Reply:
Good idea! Changed in https://github.com/Automattic/woocommerce-payments/pull/1840/commits/c9b58c63b20bfbb86a8f2888960c2dd21e399a46